client/dfclient/include/text_creator.hpp

```diff
@@ -0,0 +1,118 @@
+	 * for {hold} frames and then goes down to 0 during {decay} frames, 
+	 * optionally following a function defined by {easing}. Specifying 
+	 * {decay} as < 0 skips the tween creation, making the text permanently 
+	 * visible unless later modified. 
+	 * 
+	 * Every parameter (besides text) can be left unspecified via omission or 
+	 * std::nullopt to use previously set defaults. 
+	 * 
+	 * Texts created are scaled with regard to window width.
+	 * 
+	 * @param text The displayed text value. The only mandatory argument.
+	 * @param color The color of the text. Not including alpha, which is set 
+	 * 	via <from>.
+	 * @param pos_x The x position of the text center, measured in pixels.
+	 * @param pos_y The y posiion of the text center, measured in pixels.
+	 * @param scale Scaling factor (1.0f is default value).
+	 * @param from The starting alpha value (0-255).
+	 * @param hold The number of frames to hold the initial alpha value.
+	 * @param decay The number of frames to take to reduce alpha to 0. Set 
+	 * 	as < 0 to skip tween creation.
+	 * @param layer A layer from Layers enum to draw the text on.
+	 * @param easing The tween easing function. For more details, see 
+	 * 	the tweeny::via() definition.
+	 * @return An std::unique_ptr< ncine::TextNode > of the created text.
+	 */
+	template <typename T = tweeny::easing::linearEasing>
+	std::unique_ptr<ncine::TextNode> CreateText(
+		std::string text,
+		std::optional<ncine::Color> color = std::nullopt,
+		std::optional<float> pos_x = std::nullopt,
+		std::optional<float> pos_y = std::nullopt,
+		std::optional<float> scale = std::nullopt,
+		std::optional<int> from = std::nullopt,
+		std::optional<int> hold = std::nullopt,
+		std::optional<int> decay = std::nullopt,
+		std::optional<Layers> layer = std::nullopt,
+		const T& easing = tweeny::easing::linear
+		);
+
+	/** Create an outline of the last text. 
+	 * 
+	 * Creates an ncine::TextNode using the saved parameters of the most recently 
+	 * created text and the outline font. The only parameter that must be additionally 
+	 * specified (if not default) is the easing.
+	 * 
+	 * @param easing The easing used in the original text's tween.
+	 * @return An std::unique_ptr< ncine::TextNode > of the text outline.
+	 */
+	template <typename T = tweeny::easing::linearEasing>
+	std::unique_ptr<ncine::TextNode> CreateOutline(const T& easing = tweeny::easing::linear);
+
+	template <typename T = tweeny::easing::linearEasing>
+	tweeny::tween<int> CreateTextTween(ncine::TextNode* textPtr, int from = 255, int hold = 60, int decay = 60, const T& easing = tweeny::easing::linear);
+
+private:
+	template <typename T>
+	std::unique_ptr<ncine::TextNode> doCreateText(std::string text, ncine::Color color,
+		float pos_x, float pos_y, float scale, int from, int hold, int decay, Layers layer, const T& easing, bool isOutline);
+};
+
+#include "text_creator_templates.hpp"
+
+extern std::optional<TextCreator> textCreator;
```

Comment:
Do we really need a global...?

Reply:
We actually do not. I thought it would be a good idea, but after consideration it is not. I will remove this.